Hazel.UnitTests/Dtls/DtlsConnectionTests.cs

```diff
@@ -727,6 +727,21 @@ public void DtlsIPv4ConnectionTest()
             }
         }
 
+        [TestMethod]
+        public void DtlsSessionV0ConnectionTest()
+        {
+            using (ThreadLimitedUdpConnectionListener listener = this.CreateListener(2, new IPEndPoint(IPAddress.Any, 4296), new TestLogger()))
+            using (DtlsUnityConnection connection = this.CreateConnection(new IPEndPoint(IPAddress.Loopback, 4296), new TestLogger()))
+            {
+                connection.HazelSessionVersion = 0;
```

Comment:
DTLS PeerInfo is never bubbled up into Hazel, so it's hard to verify that this even does something, but if we assume that this setting does change what session values we send (pretty easily verified) then you can double check by seeing in the logs that this connection sends just one Cert record while the majority of other tests send two.
Mildly interesting that the MalformedClientHello tests reimplement sending client hello from the old code and thus also send version 0 session.

Reply:
Net result is these tests suck and are just for smoke anyway.

Reply:
Concurred - yeah I think this is a solid integration test that covers the interaction of a few moving parts! We probably could get more granular and add more unit tests - I think that could be good future hardening tasks when we have the bandwidth